plugins/json/json.cxx

```diff
@@ -0,0 +1,392 @@
+			ctx.callbacks().add_data_callback([this](const string& data_name, Ref ref) { this->write_data(data_name, ref); }, data_path_pair.first);
+		}
+
+		ctx.logger().info("Plugin loaded successfully");
+	}
+
+	~json_plugin() { context().logger().info("Closing plugin"); }
+
+
+private:
+	/** Read the configuration file
+	 *
+	 * \param logger PDI's logger instance
+	 * \param spec_tree the yaml tree
+	 */
+	void read_config_tree(Logger& logger, PC_tree_t spec_tree)
+	{
+		if (PC_status(spec_tree)) {
+			logger.error("Error in read_config_tree");
+			return;
+		}
+
+		// Read yaml entries
+		opt_each(spec_tree, [&](PC_tree_t elem_tree) {
+			if (to_string(PC_get(elem_tree, "{%d}", 0)) == "file") {
+				Expression filepath = to_string(PC_get(elem_tree, ".file"));
+
+				Expression default_when = 1L;
+				each(elem_tree, [&](PC_tree_t key_tree, PC_tree_t value_tree) {
+					string key = to_string(key_tree);
+
+					if (key == "when") {
+						default_when = to_string(value_tree);
+					} else if (key != "file" && key != "write") {
+						throw Config_error{key_tree, "Unknown keyword '{}' encountered while expecting file, when, or write", key};
+					}
+				});
+
+				each(elem_tree, [&](PC_tree_t key_tree, PC_tree_t value_tree) {
+					string key = to_string(key_tree);
+
+					if (key == "write") {
+						PC_tree_t write_tree = PC_get(elem_tree, ".write");
+
+						if (!PC_status(PC_get(write_tree, "[0]"))) { // it's a list of names only
+							each(write_tree, [&](PC_tree_t tree) {
+								string dset_string = to_string(tree);
+
+								// Append to list if key exist, else create it
+								auto iter = m_data_to_path_map.find(dset_string);
+								if (iter != m_data_to_path_map.end()) {
+									iter->second.push_back(make_pair(default_when, filepath));
+								} else {
+									m_data_to_path_map.insert(
+										make_pair(dset_string, std::vector<pair<Expression, Expression>>{make_pair(default_when, filepath)})
+									);
+								}
+							});
+						} else {
+							throw Config_error{key_tree, "Unknown write method. Please use [var1, var2, ...]"};
+						}
+					}
+				});
+			} else { // it's "var: filename" format
+				string dset_string = to_string(PC_get(elem_tree, "{%d}", 0));
+				Expression filepath = to_string(PC_get(elem_tree, "<%d>", 0));
+
+				auto iter = m_data_to_path_map.find(dset_string);
+				Expression default_when = 1L;
+				if (iter != m_data_to_path_map.end()) {
+					iter->second.push_back(make_pair(default_when, filepath));
+				} else {
+					m_data_to_path_map.insert(make_pair(dset_string, std::vector<pair<Expression, Expression>>{make_pair(default_when, filepath)}));
+				}
+			}
+		});
+	}
+
+	/** Write to json a scalar data 
+	 *
+	 * \param json_data A json data to which we write a scalar data
+	 * \param reference A reference to a scalar datatype
+	 * \param logger PDI's logger instance
+	 */
+	void write_scalar_to_json(nlohmann::json& json_data, Ref_r reference, Logger& logger)
```

Comment:
It is not used for the moment. I kept the logger in case we want to put some debug info for example.

Reply:
> On certain compilers, this may produce an unused variable warning. Best would be to remove the logger if it is unused. It is easy to add back if needed.

Logger removed in a recent commit